packages/nc-gui/composables/useColumnCreateStore.ts

```diff
@@ -215,6 +215,11 @@ const [useProvideColumnCreateStore, useColumnCreateStore] = createInjectionState
     const addOrUpdate = async (onSuccess: () => void, columnPosition?: Pick<ColumnReqType, 'column_order'>) => {
       try {
         if (!(await validate())) return
+        const specialCharsRegex = /[!@#$%^&*(),.?":{}|<>]/
+        if (specialCharsRegex.test(formState._rawValue.column_name)) {
+          message.warning('Special characters are not allowed in the column name.')
+          return
+        }
```

Comment:
I'd prefer to put the logic in `validators` (see L78) so that the invalid case will return in L217. Also please think about the backend side, like I can still input special characters using API.